x-pack/legacy/plugins/xpack_main/public/services/xpack_info.js

```diff
@@ -11,68 +11,66 @@ import { convertKeysToCamelCaseDeep } from '../../../../server/lib/key_case_conv
+    this.setAll(initialInfo);
+  }
 
-    setAll = (updatedXPackInfo) => {
-      // The decision to convert kebab-case/snake-case keys to camel-case keys stemmed from an old
-      // convention of using kebabe-case/snake-case in API response bodies but camel-case in JS
-      // objects. See pull #29304 for more info.
-      const camelCasedXPackInfo = convertKeysToCamelCaseDeep(updatedXPackInfo);
-      $window.sessionStorage.setItem(XPACK_INFO_KEY, JSON.stringify(camelCasedXPackInfo));
-    };
+  get = (path, defaultValue = undefined) => {
+    const xpackInfoValuesJson = window.sessionStorage.getItem(XPACK_INFO_KEY);
+    const xpackInfoValues = xpackInfoValuesJson ? JSON.parse(xpackInfoValuesJson) : {};
+    return get(xpackInfoValues, path, defaultValue);
+  };
 
-    clear = () => {
-      $window.sessionStorage.removeItem(XPACK_INFO_KEY);
-    };
+  setAll = (updatedXPackInfo) => {
+    // The decision to convert kebab-case/snake-case keys to camel-case keys stemmed from an old
+    // convention of using kebabe-case/snake-case in API response bodies but camel-case in JS
+    // objects. See pull #29304 for more info.
+    const camelCasedXPackInfo = convertKeysToCamelCaseDeep(updatedXPackInfo);
+    // guarding sessionStorage for testing
+    typeof sessionStorage !== 'undefined' && sessionStorage.setItem(XPACK_INFO_KEY, JSON.stringify(camelCasedXPackInfo));
+  };
 
-    refresh = () => {
-      if (this.inProgressRefreshPromise) {
-        return this.inProgressRefreshPromise;
-      }
+  clear = () => {
+    sessionStorage.removeItem(XPACK_INFO_KEY);
+  };
 
-      // store the promise in a shared location so that calls to
-      // refresh() before this is complete will get the same promise
-      const $http = $injector.get('$http');
-      this.inProgressRefreshPromise = (
-        $http.get(chrome.addBasePath('/api/xpack/v1/info'))
-          .catch((err) => {
-          // if we are unable to fetch the updated info, we should
-          // prevent reusing stale info
-            this.clear();
-            xpackInfoSignature.clear();
-            throw err;
-          })
-          .then((xpackInfoResponse) => {
-            this.setAll(xpackInfoResponse.data);
-            xpackInfoSignature.set(xpackInfoResponse.headers('kbn-xpack-sig'));
-          })
-          .finally(() => {
-            this.inProgressRefreshPromise = null;
-          })
-      );
+  refresh = $injector => {
+    if (this.inProgressRefreshPromise) {
       return this.inProgressRefreshPromise;
-    };
+    }
 
-    getLicense = () => {
-      return this.get('license', {
-        isActive: false,
-        type: undefined,
-        expiryDateInMillis: undefined,
-      });
-    };
-  }
+    // store the promise in a shared location so that calls to
+    // refresh() before this is complete will get the same promise
+    const $http = $injector.get('$http');
+    this.inProgressRefreshPromise = (
+      $http.get(chrome.addBasePath('/api/xpack/v1/info'))
```

Comment:
use async/await and replace with kfetch
```
const response = await kfetch({ pathname: '/api/xpack/v1/info' });
```

Reply:
(base path is added automatically with kfetch)

Reply:
`import { kfetch } from '../../../../ui/public/kfetch';`

Reply:
I suspect this is the obstacle to using kfetch - https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/xpack_main/public/hacks/check_xpack_info_change.js#L107

```
module.config(($httpProvider) => {
  $httpProvider.interceptors.push('checkXPackInfoChange');
});
```

Reply:
I also tried to inject $http directly but that creates circular dependency errors.